corehq/apps/analytics/static/analytix/js/logging.js

```diff
@@ -32,24 +32,24 @@ hqDefine('analytix/js/logging', [
         } else if (_.isArguments(message.value)) {
             _title = "Arguments (" + message.value.length + ")    " + _.map(Array.from(message.value), JSON.stringify).join('    ');
             _value = Array.from(message.value);
-            _group = new Group(_title, new Message(_value, message.level));
+            _group = Group(_title, Message(_value, message.level));
```

Comment:
The convention is to name classes that require the `new` operator with `UpperCaseNames` and functional-style with `camelCaseNames`. I think we should preserve that convention if possible (even though we will hopefully eventually have no `UpperCaseNames` in our codebase). In this case `Group` should be renamed to `group`.

Reply:
That's Crockford's convention, but is there a good case for also making it ours? I think we currently do `UpperCaseNames` for everything class-like.

Reply:
I like the convention because it makes it easy to tell when you should be using `new`, but just forgot. If we don't use the convention then it requires a lot deeper searching and/or knowledge of context for code reviewers to know if something is wrong with the code.

A counter-argument could be that we already violate it so much that it's not a good indicator in our code base, but that seems like dirty style that should be cleaned up rather than a good argument. Feel free to push back.

Reply:
I do agree it would be good for the casing to mean something, and I'm definitely biased by not wanting to do naming changes at the same time as removing `new`. But more generally, casing doesn't mean anything in our js. We don't even agree on `camelCase` versus `snake_case`, even within single files*. That haphazard-ness makes me skeptical that the uppercase/lowercase distinction will be noticeable.

* To be fair, casing does get ambiguous when js pulls data from python, which adheres to python conventions...which is part of why I haven't made the effort to champion better naming conventions.

Reply:
I'll defer to you on what you think is possible given the current state of things. What do you think of these guidelines:

When adding new code:
- Use `camelCase` for all variable and function names.
- Use `TitleCase` for class names that must be instantiated with the `new` operator (but avoid using the `new` operator if at all possible).

Reply:
should these new guidelines be communicated more broadly if people weren't following this conversation?

Reply:
yeah, good point, will update js guide